app/models/encounter.rb

```diff
@@ -0,0 +1,42 @@
+class Encounter < ApplicationRecord
+  belongs_to :patient
+  belongs_to :facility
+
+  has_many :encounter_events
+  has_many :blood_pressures, through: :encounter_events, source: :encounterable, source_type: 'BloodPressure'
+
+  def events
```

Comment:
Again, these should probably be `observations`. The language in this method is not obvious.